### ISSUE TITLE
Clarification about main_Semantic3D 

### ISSUE THREAD

**Issue by Gabry993** [QingyongHu/RandLA-Net#45]
Hi @QingyongHu,
first of all, thank you for sharing your work. I managed to train the model with my own data achieving promising results!
I'm trying to implement your code in pytorch for my convenience, so maybe this won't be my last doubt :P
Looking at line 208 in `main_Semantic3D.py` you do this thing:
`queried_pc_xyz[:, 0:2] = queried_pc_xyz[:, 0:2] - pick_point[:, 0:2]`
I would say that you normalize the `x` and `y` coordinates of the queried points with respect to queried center. Why do you leave `z` out? Is there any particular reason I'm missing?
Thank you,
Gabry

**Comment by QingyongHu** [comment]
Hi @Gabry993, thanks for your interest in our work!

This is mainly because we feel that the height information would be helpful to separate points from different categories. For example, using height information only is able to roughly separate the ground and trees. Therefore, we keep the original z-axis information to achieve better segmentation performance.

**Comment by Gabry993** [comment]
Ok, I see. Thank you for your quick response :)

**Comment by Gabry993** [comment]
I managed to implement it in pytorch and I got the same great performances on my dataset. Again, thank you for sharing your work :)

**Comment by 123zhen123** [comment]
> I managed to implement it in pytorch and I got the same great performances on my dataset. Again, thank you for sharing your work :)

Hi, I am now confused because there is no pytorch implementation version. Can you share your pytorch version for reference and study, I would be very grateful :)

**Comment by Gabry993** [comment]
@123zhen123 Sorry for getting back to you so late.
I've been re-implementing the pytorch model for a larger project and I was able to "extract" it only recently. You can find it [here](https://github.com/idsia-robotics/RandLA-Net-pytorch).
Note that it was bent to our needs and data formats, but all the relevant pytorch code is there and adapting it to something different should be pretty easy. The idea has been to stay as close as possible to the tensorflow code. We tested the original implementation on our data and we were able to get the same results with the pytorch one and this was enough for our needs.
 Unfortunately I don't have time to work on this anymore or document it better.
